### PR TITLE
Modify the `infra test cargo` command to run more tests.

### PR DESCRIPTION
**Pull request by mjoerussell**
I discovered this while looking into #1181. I tried adding a test case to the query parsing tests, but found that none of them were getting run. This PR fixes only that issue, nothing else about #1181.

Previously, the command used to run tests was `cargo nextest run --workspace --all-features --lib --bins --examples --no-fail-fast`. This command ended up skipping the tests in the `metaslang` packages. We can run these as well by changing the command to `cargo nextest run --workspace --exclude solidity_testing_perf --all-features --all-targets --no-fail-fast`.

`--exclude solidity_testing_perf` is required because it tries to link with callgrind, which will not be available on all platforms. It's possible that this crash was the reason for the original restrictions in the first place, but I don't know for sure. We can think about different ways to conditionally run those tests in the future, if it's wanted.

### CODE REVIEW

File: crates/infra/cli/src/commands/test/mod.rs [hunk 1]
```diff
@@ -44,11 +44,10 @@ fn test_cargo() -> Result<()> {
 
     Command::new("cargo")
```
Comment (OmarTawfik) [inline crates/infra/cli/src/commands/test/mod.rs:45]: Great catch! Thank you!

`solidity_testing_perf` contains two kinds of tests, the benchmarks (which need `valgrind`), and the the same logic ran as unit tests (without measuring) to ensure they are correct in CI. So I think this would disable both kinds.

I tried using [per test overrides](https://nexte.st/docs/configuration/per-test-overrides/) to exclude only the benchmark tests, but the test runner fails due to an external issue ([I filed a bug here](https://github.com/iai-callgrind/iai-callgrind/issues/339)).

In the meantime, WDYT of adding a second explicit command after it, to make sure both run in CI?

```bash
# Run all tests in 'solidity_testing_perf' excluding the benchmarks (which need valgrind installed on the system):
cargo nextest run --package "solidity_testing_perf" --workspace --all-features --lib --bins --examples --no-fail-fast
```

**Comment by ggiraldez** [comment]
I think you want to add `--tests` instead, which compiles the source files of each package as tests. The difference is that `--all-targets` also includes `--benches` which is what we want to exclude from normal testing (and why you need to exclude `solidity_testing_perf`).

File: crates/infra/cli/src/commands/test/mod.rs [hunk 2]
```diff
@@ -44,11 +44,10 @@ fn test_cargo() -> Result<()> {
...
         .args(["nextest", "run"])
+        .args(["--exclude", "solidity_testing_perf"]) // Requires callgrind, which may not be available
         .flag("--workspace")
         .flag("--all-features")
-        .flag("--lib")
-        .flag("--bins")
-        .flag("--examples")
+        .flag("--all-targets")
```
Comment (ggiraldez) [inline crates/infra/cli/src/commands/test/mod.rs:50]: I think you want to add `--tests` instead, which compiles the source files of each package as tests. The difference is that `--all-targets` also includes `--benches` which is what we want to exclude from normal testing (and why you need to exclude `solidity_testing_perf`).

**Comment by mjoerussell** [comment]
> In the meantime, WDYT of adding a second explicit command after it, to make sure both run in CI?

I believe that the original command was already skipping these perf tests, so would we want to change the behavior here? Isn't it enough to run them in `infra perf`?

**Comment by OmarTawfik** [comment]
> I believe that the original command was already skipping these perf tests, so would we want to change the behavior here? Isn't it enough to run them in infra perf?

It is not, since `infra perf` doesn't run in the `ci.yml` workflow on PRs. It only runs in the `benchmark.yml` workflow on `main` branch builds. There is a small amount of additional work that needs to happen to enable `benchmark.yml` in PRs first:

- https://bencher.dev/docs/how-to/github-actions/#pull-requests
- https://bencher.dev/docs/how-to/github-actions/#pull-requests-from-forks

**Comment by ggiraldez** [comment]
> > I believe that the original command was already skipping these perf tests, so would we want to change the behavior here? Isn't it enough to run them in infra perf?
> 
> It is not, since `infra perf` doesn't run in the `ci.yml` workflow on PRs. It only runs in the `benchmark.yml` workflow on `main` branch builds. There is a small amount of additional work that needs to happen to enable `benchmark.yml` in PRs first:
> 
> * https://bencher.dev/docs/how-to/github-actions/#pull-requests
> * https://bencher.dev/docs/how-to/github-actions/#pull-requests-from-forks

The original command did include the unit tests in perf, but only adding `--tests` does not exclude them (as the package no longer needs to be excluded). So, I don't think an extra command is necessary.

File: crates/infra/cli/src/commands/test/mod.rs [hunk 3]
```diff
@@ -46,9 +46,7 @@ fn test_cargo() -> Result<()> {
         .args(["nextest", "run"])
         .flag("--workspace")
         .flag("--all-features")
-        .flag("--lib")
-        .flag("--bins")
-        .flag("--examples")
```
Comment (ggiraldez) [inline crates/infra/cli/src/commands/test/mod.rs:51]: I'm not sure we want to remove these other flags. While `--tests` *by default* will include all of these, they can be disabled via `test = false` in the Cargo manifest.

**Review by OmarTawfik — CHANGES_REQUESTED** [review]
passing back to @mjoerussell, based on @ggiraldez's suggestions above.

**Review by OmarTawfik — APPROVED** [review]
